registry/rpc/rpcregistry.go

```diff
@@ -73,7 +73,7 @@ func (r *RPCRegistry) getClient() pb.RegistryClient {
 func (r *RPCRegistry) Connect() {
 	// We want the connection operation to block and constantly reconnect using grpc backoff
 	log.Info("Starting gRPC connection to fleet-engine...")
-	connection, err := grpc.Dial(":fleet-engine:", grpc.WithInsecure(), grpc.WithDialer(r.dialer), grpc.WithBlock())
+	connection, err := grpc.Dial(":fleet-engine:", grpc.WithTimeout(12*time.Second), grpc.WithInsecure(), grpc.WithDialer(r.dialer), grpc.WithBlock())
```

Comment:
These 12seconds it is the maximum, considering the other machine is alive but not responding. We should allow sorted downtimes in order to avoid changes of leader due to `serious` issues. As an example, a quick `pkill -9 fleetd` shouldn't cause a change in the leader.